arch/xtensa/include/kernel_arch_data.h

```diff
@@ -57,6 +57,10 @@ typedef struct _kernel_arch _kernel_arch_t;
 
 #endif /*! _ASMLANGUAGE && ! __ASSEMBLER__ */
 
+#ifdef CONFIG_USE_SWITCH
```

Comment:
can we change the config name to CONFIG_USE_ARCH_SWITCH?

Reply:
why is that necessary?